### ISSUE TITLE
Proxy of private repositories

### ISSUE THREAD

**Issue by ptitvert** [dzikoysk/reposilite#278]
I wanted to use reposilite with some repo in my organisation, which are protected with some credentials.
I've added the proxies URL, but reposilite never finds them. When I use the URL to directly access the artifacts (pom or jar) on my browser, each time I need to enter my credentials (well at least the first time at the beginning of the session).

Is there a way to have such possibility to have public repo and private repo with credentials?

**Comment by dzikoysk** [comment]
Unfortunately, at this moment there is no way of providing credentials for proxied repositories. It's caused by the proxing policy - Reposilite stores all of the requested artifacts as long as it can and there is no such a possibilty to bind remote repositories to the specific repositories within the current instance. It means that a public Reposilite instance with access to the private one could easly expose secured content with a default configuration. Let's say:

```
public repository
/releases (public)
/private (private)

private repository behind the proxy
/releases (private)
/private (private)
```
As you can see, the standard `releases` repository would be exposed. We should make sure that proxied repository names won't match the public one (not secure by design and it is why it's not supported). To avoid that we need to:

* Assign proxied repositories to the specific (prefered, breaking change in proxy implementation and configuration, could be possible in next minor version -> feature request)

I've marked is a feature request, but I don't think it will be available soon due to my limited time recently.

**Comment by dzikoysk** [comment]
Addressed in #544
